cirq/sim/simulator.py

```diff
@@ -468,21 +468,30 @@ def sample_measurement_ops(
         """
         bounds = {}  # type: Dict[str, Tuple]
         all_qubits = []  # type: List[ops.Qid]
+        meas_ops = {}
         current_index = 0
         for op in measurement_ops:
             gate = op.gate
             if not isinstance(gate, ops.MeasurementGate):
                 raise ValueError('{} was not a MeasurementGate'.format(gate))
             key = protocols.measurement_key(gate)
+            meas_ops[key] = gate
             if key in bounds:
                 raise ValueError(
                     'Duplicate MeasurementGate with key {}'.format(key))
             bounds[key] = (current_index, current_index + len(op.qubits))
             all_qubits.extend(op.qubits)
             current_index += len(op.qubits)
         indexed_sample = self.sample(all_qubits, repetitions)
-        return {k: np.array([x[s:e] for x in indexed_sample]) for k, (s, e) in
-                bounds.items()}
+
+        result = {
+            k: np.array([x[s:e] for x in indexed_sample
+                        ]) for k, (s, e) in bounds.items()
+        }
+        corrected_results = {
+            k: bits ^ meas_ops[k].full_invert_mask for k, bits in result.items()
+        }
```

Comment:
Can't we combine these two?  Something like this:

```
corrected_results = {
            k: np.array([x[s:e] for x in indexed_sample]) ^ meas_ops[k].full_invert_mask for k, (s, e) in bounds.items()
        }
```

Reply:
that's way too complex for my liking.  generally if the comprehension has to span two lines I get nervous.

Reply:
That's why I suggested using a helper function.

Reply:
It looks much better now.  It was confusing when it was half comprehension, half loop.

Reply:
Yeah, I am good with that. (I think I already approved, but LGTM)

Reply:
@cduck LGTM?